.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,70 @@
+      - master
+
+jobs:
+  build-java-8:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Create the Java 8 builder image
+        # we still can't add linux/arm64/v8 because of a segfault on yarn installation on 20.04
+        run: docker buildx build -f java-8/build/Dockerfile --platform linux/amd64 -t metabase/ci:java-8-clj-1.10.1040.12-2021-build --push --compress --no-cache .
+      - name: Create the Java 8 testing image
+        run: docker buildx build -f java-8/test/Dockerfile -t metabase/ci:java-8-clj-1.10.1040.12-2021-node-browsers --push --compress --no-cache .
+    
+  build-java-11:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Create Java 11 builder image
+      # we still can't add linux/arm64/v8 because of a segfault on yarn installation on 20.04
+        run: docker buildx build -f java-11/build/Dockerfile --platform linux/amd64 -t metabase/ci:java-11-clj-1.10.1040.12-2021-build --push --compress --no-cache .
+      - name: Create Java 11 testing image
+        run: docker buildx build -f java-11/test/Dockerfile -t metabase/ci:java-11-clj-1.10.1040.12-2021-node-browsers --push --compress --no-cache .
+
+  build-java-17:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Create Java 17 builder image
+      # we still can't add linux/arm64/v8 because of a segfault on yarn installation on 20.04
+        run: docker buildx build -f java-17/build/Dockerfile --platform linux/amd64 -t metabase/ci:java-17-clj-1.10.1040.12-2021-build --push --compress --no-cache .
+      - name: Create Java 17 testing image
+        run: docker buildx build -f java-17/test/Dockerfile -t metabase/ci:java-17-clj-1.10.1040.12-2021-node-browsers --push --compress --no-cache .
```

Comment:
```suggestion
name: Build and push to Dockerhub

on:
  push:
    branches:
      - master

jobs:
  build-java:
    runs-on: ubuntu-20.04
    timeout-minutes: 30
    strategy:
      matrix:
        version: [8, 11, 17]
    steps:
      - uses: actions/checkout@v2
      - name: Set up QEMU
        uses: docker/setup-qemu-action@v1
      - name: Set up Docker Buildx
        id: buildx
        uses: docker/setup-buildx-action@v1
      - name: Login to Docker Hub
        uses: docker/login-action@v1
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
      - name: Create the Java ${{ matrix.version }} builder image
        # we still can't add linux/arm64/v8 because of a segfault on yarn installation on 20.04
        run: |
          docker buildx build
          -f java-${{ matrix.version }}/build/Dockerfile
          --platform linux/amd64
          -t metabase/ci:java-${{ matrix.version }}-clj-1.10.1040.12-2021-build --push --compress --no-cache .
      - name: Create the Java ${{ matrix.version }} testing image
        run: |
          docker buildx build
          -f java-${{ matrix.version }}/test/Dockerfile
          -t metabase/ci:java-${{ matrix.version }}-clj-1.10.1040.12-2021-node-browsers --push --compress --no-cache .

```

Nit: why don't we use `strategy` matrix to DRY this code out a little bit?
Nit 2: Suggested giving commands their own line in the `run` step for better readability

Reply:
makes lot of sense!